gtk4/src/subclass/widget.rs

```diff
@@ -1211,12 +1211,12 @@ pub unsafe trait WidgetClassExt: ClassStruct {
 unsafe impl<T: ClassStruct> WidgetClassExt for T where T::Type: WidgetImpl {}
 
 #[derive(Debug, PartialEq, Eq)]
-#[repr(transparent)]
```

Comment:
I hope nothing was relying on it being the same representation as `T` :)

Reply:
Maybe this needs to be repr-C now so that we at least guarantee that the pointer is stored at the beginning? How is GtkBuilder filling this?

Reply:
> Maybe this needs to be repr-C now so that we at least guarantee that the pointer is stored at the beginning? How is GtkBuilder filling this?

By calling gtk_widget_class_bind_template_child_full, which is used from rust with bind_template_child_with_offset. But yes, maybe it should be repr-C.

Reply:
Yes, it currently works with the offset to `TemplateChild` so you need to make it repr-C at least

Reply:
Done, let us wait and see if @felinira got whatever use case they have working with this patch first. I tested it locally by changing one of the examples but still

Reply:
> I'm a little surprised why this needs a FromValue implementation though. Isn't this readonly?

The requirement seems to come from `ObjectExt::property`
```rust
fn property<V: for<'b> FromValue<'b> + 'static>(&self, property_name: &str) -> V;
```

So it is expected? not sure why you are surprised about it, unless you meant the ToValue? in such case, FromValue, requires ValueType, which itself requires ToValue....